aws/data_source_aws_cloudfront_function_test.go

```diff
@@ -0,0 +1,65 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/cloudfront"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataSourceAWSCloudfrontFunction_basic(t *testing.T) {
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	dataSourceName := "data.aws_cloudfront_function.test"
+	resourceName := "aws_cloudfront_function.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:   func() { testAccPreCheck(t) },
+		ErrorCheck: testAccErrorCheck(t, cloudfront.EndpointsID),
+		Providers:  testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAWSCloudfrontFunctionConfigBasic(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrPair(dataSourceName, "arn", resourceName, "arn"),
+					resource.TestCheckResourceAttrPair(dataSourceName, "code", resourceName, "code"),
+					resource.TestCheckResourceAttrPair(dataSourceName, "comment", resourceName, "comment"),
+					resource.TestCheckResourceAttrPair(dataSourceName, "last_modified", resourceName, "last_modified"),
```

Comment:
I got

```
=== CONT  TestAccDataSourceAWSCloudfrontFunction_basic
    data_source_aws_cloudfront_function_test.go:17: Step 1/1 error: Check failed: 1 error occurred:
        	* Check 4/9 error: data.aws_cloudfront_function.test: Attribute 'last_modified' expected "2021-05-13T14:12:49Z", got "2021-05-13T14:12:48Z"
```

that should pass once setting `last_modified` is moved from resource `Create` to resource `Read` (I think the `Publish` is called one second later 😄).

Reply:
Yeah, sometimes I've got the same. But it doesn't make any sense. 
I must be some quirk in AWS.